source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -741,101 +741,55 @@ namespace
-            if (input->hasNodeName())
+            for (InputPtr input : surfaceShader->getActiveInputs())
             {
-                if (input->hasOutputString())
+                BindInputPtr bindInput = shaderRef->addBindInput(input->getName(), input->getType());
+                if (input->hasNodeName())
+                {
+                    if (input->hasOutputString())
+                    {
+                        bindInput->setNodeGraphString(input->getNodeName());
+                        bindInput->setOutputString(input->getOutputString());
+                    }
+                }
+                else
                 {
-                    bindInput->setNodeGraphString(input->getNodeName());
-                    bindInput->setOutputString(input->getOutputString());
+                    bindInput->setValueString(input->getValueString());
                 }
             }
-            else
+            for (ParameterPtr param : surfaceShader->getActiveParameters())
             {
-                bindInput->setValueString(input->getValueString());
+                BindParamPtr bindParam = shaderRef->addBindParam(param->getName(), param->getType());
+                bindParam->setValueString(param->getValueString());
             }
-        }
-        for (ParameterPtr param : mxNode->getActiveParameters())
-        {
-            BindParamPtr bindParam = shaderRef->addBindParam(param->getName(), param->getType());
-            bindParam->setValueString(param->getValueString());
-        }
-        // Should we delete the surface shader?
-        if (writeOptions->materialWriteOp & RtWriteOptions::MaterialWriteOp::DELETE)
-        {
-            doc->removeChild(prim->getName());
-        }
-        // Should we create a look for the material element?
-        if (writeOptions->materialWriteOp & RtWriteOptions::MaterialWriteOp::LOOK)
-        {
-            LookPtr look = doc->addLook();
-            MaterialAssignPtr materialAssign = look->addMaterialAssign();
-            materialAssign->setMaterial(material->getName());
-            CollectionPtr collection = doc->addCollection();
-            collection->setIncludeGeom("/*");
-            materialAssign->setCollection(collection);
-        }
-    }
 
-    void writeMaterialElements(const PvtPrim* prim, NodePtr mxNode, DocumentPtr doc, const RtWriteOptions* writeOptions)
-    {
-        if (writeOptions->materialWriteOp & RtWriteOptions::MaterialWriteOp::ADD_MATERIAL_NODES_FOR_SHADERS)
-        {
-            // Get the connected material nodes and create material elements from them (using their names)
-            const PvtOutput* output = prim->getOutput(PvtAttribute::DEFAULT_OUTPUT_NAME);
-            if (output->isConnected())
+            // Should we create a look for the material element?
+            if (writeOptions->materialWriteOp & RtWriteOptions::MaterialWriteOp::LOOK)
```

Comment:
Jerry was confused about the logic for the opposite case of this which is to not explicitly create looks. 
Can you rename ::LOOK to CREATE_LOOK(S) or something like that? Thanks